components/backup-stream/src/metrics.rs

```diff
@@ -3,14 +3,22 @@
 use lazy_static::lazy_static;
 use prometheus::*;
 
+/// The status of a task.
+/// The ordering of this imples the priority for presenting to the user.
+/// max(TASK_STATUS) of all stores would be probably the state of the task.
+#[derive(Debug, Clone, Copy)]
 pub enum TaskStatus {
     Running = 0,
     Paused,
     Error,
 }
 
 pub fn update_task_status(status: TaskStatus, task: &str) {
-    TASK_STATUS.with_label_values(&[task]).set(status as _);
+    let g = TASK_STATUS.with_label_values(&[task]);
+    // If the current status is greater than now, do nothing.
+    if g.get() < status as _ {
```

Comment:
```suggestion
    if g.get() < status as _ && status != TaskStatus::Running {
```
This would make us cannot reset the status to NORMAL, the allowed status transforming should be:
```
┌─────────┐      ┌────────┐      ┌───────┐
│         │      │        │      │       │
│ RUNNING ├──────► PAUSED ├──────► ERROR │
│         │      │        │      │       │
└────▲────┘      └────────┘      └───┬───┘
     │                               │
     │                               │
     └───────────────────────────────┘
```